source/agora/consensus/protocol/Nominator.d

```diff
@@ -983,8 +983,9 @@ extern(D):
         // new network connections might need to be established).
         if (!this.acceptBlock(signed_block))
         {
-            log.error("Block was not accepted by node {}", this.kp.address);
-            assert(0, format!"Block was not accepted"());
+            auto msg = this.ledger.validateBlock(signed_block);
```

Comment:
The `acceptBlock` already has the following code. Is calling the `validateBlock` in the above code just for showing the error message explicitly again?
```
        if (auto fail_reason = this.validateBlock(block, file, line))
        {
            log.trace("Rejected block: {}: {}", fail_reason, block.prettify());
            return false;
        }
```

Reply:
Yep, because at the moment, the message might not get printed because the level is `Trace`.

Reply:
To be precise, the nodes stop responding to any request because they just crash after generating an invalid block, which means the regular log-dumping done by our `assert` handler doesn't happen (it's triggered from the main thread).